presto-memory/src/test/java/io/prestosql/plugin/memory/TestMemorySmoke.java

```diff
@@ -83,6 +91,49 @@ public void testSelect()
         assertQueryResult("SELECT count(*) FROM test_select", 75L);
     }
 
+    @Test
+    public void testJoinDynamicFilteringBlockProbeSide()
+    {
+        Session session = Session.builder(getSession())
+                .setSystemProperty(ENABLE_DYNAMIC_FILTERING, "true")
+                .setSystemProperty(JOIN_DISTRIBUTION_TYPE, FeaturesConfig.JoinDistributionType.BROADCAST.name())
+                .setSystemProperty(PUSH_PARTIAL_AGGREGATION_THROUGH_JOIN, "true")
+                .build();
+        DistributedQueryRunner runner = (DistributedQueryRunner) getQueryRunner();
+        ResultWithQueryId<MaterializedResult> result = runner.executeWithQueryId(session, "SELECT MAX(quantity) FROM lineitem JOIN orders " +
```

Comment:
Not sure I understand the suggestion - if I use the following query:
```sql
SELECT * 
FROM (SELECT MAX(quantity), orderkey FROM lineitem GROUP BY orderkey) l
JOIN orders 
ON   l.orderkey = orders.orderkey;
```
It results in a distributed join (due to a repartition operator between the TableScan of `lineitem` and `MAX` aggregation:
```
 Fragment 1 [HASH]                                                                                                                                                                                          
     Output layout: [max, orderkey, custkey, orderstatus, totalprice, orderdate, orderpriority, clerk, shippriority, comment_10]                                                                            
     Output partitioning: SINGLE []                                                                                                                                                                         
     Stage Execution Strategy: UNGROUPED_EXECUTION                                                                                                                                                          
     InnerJoin[("orderkey" = "orderkey_9")][$hashvalue_59, $hashvalue_60]                                                                                                                                   
     │   Layout: [orderkey:bigint, max:double, custkey:bigint, orderstatus:varchar(1), totalprice:double, orderdate:date, orderpriority:varchar(15), clerk:varchar(15), shippriority:integer, comment_10:var
     │   Estimates: {rows: ? (?), cpu: ?, memory: ?, network: ?}                                                                                                                                            
     │   Distribution: PARTITIONED                                                                                                                                                                          
     │   dynamicFilterAssignments = {orderkey_9 -> df_283}                                                                                                                                                  
     ├─ Project[]                                                                                                                                                                                           
     │  │   Layout: [orderkey:bigint, max:double, $hashvalue_59:bigint]                                                                                                                                     
     │  │   Estimates: {rows: ? (?), cpu: ?, memory: ?, network: ?}                                                                                                                                         
     │  │   $hashvalue_59 := combine_hash(bigint '0', COALESCE("$operator$hash_code"("orderkey"), 0))                                                                                                       
     │  └─ Aggregate[orderkey]                                                                                                                                                                              
     │     │   Layout: [orderkey:bigint, max:double]                                                                                                                                                        
     │     │   Estimates: {rows: ? (?), cpu: ?, memory: ?, network: ?}                                                                                                                                      
     │     │   max := max("quantity")                                                                                                                                                                       
     │     └─ LocalExchange[HASH][$hashvalue] ("orderkey")                                                                                                                                                  
     │        │   Layout: [orderkey:bigint, quantity:double, $hashvalue:bigint]                                                                                                                             
     │        │   Estimates: {rows: ? (?), cpu: ?, memory: 0B, network: ?}                                                                                                                                  
     │        └─ RemoteSource[2]                                                                                                                                                                            
     │               Layout: [orderkey:bigint, quantity:double, $hashvalue_57:bigint]                                                                                                                       
     └─ LocalExchange[HASH][$hashvalue_60] ("orderkey_9")                                                                                                                                                   
        │   Layout: [orderkey_9:bigint, custkey:bigint, orderstatus:varchar(1), totalprice:double, orderdate:date, orderpriority:varchar(15), clerk:varchar(15), shippriority:integer, comment_10:varchar(79
        │   Estimates: {rows: ? (?), cpu: ?, memory: 0B, network: ?}                                                                                                                                        
        └─ RemoteSource[3]                                                                                                                                                                                  
               Layout: [orderkey_9:bigint, custkey:bigint, orderstatus:varchar(1), totalprice:double, orderdate:date, orderpriority:varchar(15), clerk:varchar(15), shippriority:integer, comment_10:varchar
```
But then, local dynamic filtering doesn't take place - because it needs broadcast join...